File: SERVER/src/model/booking.js

const mongoose = require('mongoose')
const Booking = mongoose.model('booking', {
    bookingdate: {
        type: String,
        trim: true,
    },
    bookingid: {
        type: String,
        required: true,
    },
    checkin: {
        type:String,
        required: true,

    },
    checkout: {
        type: String,
        required: true,

    },
    bookingprice: {
        type: Number,
    },
    guests: {
        type: Number,
        required: true,
    },
    numberofdays: {
        type: Number,
    },
    userid: {
        type: String,
    },
    houseid: {
        type: String,
    },
})



module.exports = Booking;

File: lib/models/booking.dart
import 'package:json_annotation/json_annotation.dart';

part 'booking.g.dart';
@JsonSerializable(explicitToJson: true)
class Booking {
  String? bookingdate;
  String? bookingid;
  DateTime? checkin;
  DateTime? checkout;
  double? bookingprice;
  int? guests;
  int? numberofdays;
  String? userid;
  String? houseid;

  Booking({
    required this.bookingdate,
    required this.bookingid,
    required this.checkin,
    required this.checkout,
    required this.bookingprice,
    required this.guests,
    required this.numberofdays,
    required this.userid,
    required this.houseid,
  });

 factory Booking.fromJson(Map<String, dynamic> json) =>_$BookingFromJson(json);

  Map<String, dynamic> toJson() => _$BookingToJson(this);
}
